core/src/main/java/com/datastrato/gravitino/listener/impl/EventListenerManager.java

```diff
@@ -0,0 +1,154 @@
+                            MapUtils.getPrefixMap(properties, DOT.join(listenerName, ""))),
+                    (existingValue, newValue) -> {
+                      throw new IllegalStateException(
+                          "Duplicate event listener name detected: " + existingValue);
+                    }));
+    this.eventListeners = assembleEventListeners(userEventListenerPlugins);
+  }
+
+  public void start() {
+    eventListeners.stream().forEach(listener -> listener.start());
+  }
+
+  public void stop() {
+    eventListeners.stream().forEach(listener -> listener.stop());
+  }
+
+  public EventBus createEventBus() {
+    return new EventBus(eventListeners);
+  }
+
+  private List<EventListenerPlugin> assembleEventListeners(
+      Map<String, EventListenerPlugin> userEventListeners) {
+    List<EventListenerPlugin> sharedQueueListeners = new ArrayList<>();
+
+    List<EventListenerPlugin> listeners =
+        userEventListeners.entrySet().stream()
+            .map(
+                entrySet -> {
+                  String listenerName = entrySet.getKey();
+                  EventListenerPlugin listener = entrySet.getValue();
+                  if (listener instanceof SupportsAsync) {
+                    SupportsAsync asyncListener = (SupportsAsync) listener;
+                    if (Mode.SHARED.equals(asyncListener.asyncMode())) {
+                      sharedQueueListeners.add(
+                          new EventListenerPluginWrapper(listenerName, listener));
+                      return null;
+                    } else {
+                      return new AsyncQueueListener(
+                          ImmutableList.of(new EventListenerPluginWrapper(listenerName, listener)),
+                          listenerName,
+                          queueCapacity,
+                          dispatcherJoinSeconds);
+                    }
+                  } else {
+                    return new EventListenerPluginWrapper(listenerName, listener);
+                  }
+                })
+            .filter(listener -> listener != null)
+            .collect(Collectors.toList());
+
+    if (sharedQueueListeners.size() > 0) {
+      listeners.add(
+          new AsyncQueueListener(
+              sharedQueueListeners, "default", queueCapacity, dispatcherJoinSeconds));
+    }
+    return listeners;
+  }
+
+  private EventListenerPlugin loadUserEventListenerPlugin(
+      String listenerName, Map<String, String> config) {
+    LOG.info("EventListener:{}, config:{}.", listenerName, config);
+    String className = config.get(GRAVITINO_EVENT_LISTENER_CLASSNAME);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(className),
+        String.format(
+            "EventListener:%s, %s%s.%s is not set in configuration.",
+            listenerName,
+            GRAVITINO_EVENT_LISTENER_PREFIX,
+            listenerName,
+            GRAVITINO_EVENT_LISTENER_CLASSNAME));
+
+    try {
+      EventListenerPlugin listenerPlugin =
```

Comment:
So, we won't allow third-party listeners currently?

Reply:
support, why not support?

Reply:
I noticed that those class are only loaded by `AppClassloader`